addons/pos_online_payment/static/src/app/store/models.js

```diff
@@ -80,7 +83,10 @@ patch(Order.prototype, {
         }
         for (const missingInServerLine of opLinesToUpdate) {
             if (missingInServerLine.get_payment_status() === "done") {
-                this.paymentlines.remove(missingInServerLine);
+                this.paymentlines = this.paymentlines.filter(
+                    (l) => l.cid !== missingInServerLine.cid
+                );
+
```

Comment:
`remove` is the correct action here. We should not reassign arrays because it can potentially break the reactivity (more specifically when `this` is the component`).